### PR TITLE
Better implementation to get all issue

### PR DESCRIPTION
**Pull request by TianCal**

### CODE REVIEW

File: app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java [hunk 1]
```diff
@@ -419,154 +402,78 @@ public void run() {
...
-         */
-        final List<String> list = new ArrayList<String>();
-        String credential = Credentials.basic(username, token);
+    public List<Item> getAllIssueInfo(final String username, String token) throws InterruptedException {
+        final List<Item> itemList = new ArrayList<Item>();
+        OkHttpClient client = new OkHttpClient().newBuilder()
+                .build();
+
         final CountDownLatch latch = new CountDownLatch(1);
+        String key = "MD";
```
Comment (ouyuran) [inline app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java]: Let it be const. "KEY"

File: app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java [hunk 2]
```diff
@@ -419,154 +402,78 @@ public void run() {
...
+        String credential = Credentials.basic(username, token);
         Request request = new Request.Builder()
-                .url("https://nj-summer-camp-2020.atlassian.net/rest/api/3/project")
+                .url("https://nj-summer-camp-2020.atlassian.net/rest/api/3/search?jql=project=" + key)
```
Comment (ouyuran) [inline app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java]: User url parameter method.

File: app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java [hunk 3]
```diff
@@ -419,154 +402,78 @@ public void run() {
...
-                .writeTimeout(300, TimeUnit.SECONDS)
-                .readTimeout(300, TimeUnit.SECONDS)
-                .build();
-
-        /*
-        Return a list with all the projects
-         */
 
-        client.newCall(request).enqueue(new Callback() {
+        client.newCall(request).enqueue(new Callback(){
```
Comment (ouyuran) [inline app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java:425]: space

File: app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java [hunk 4]
```diff
@@ -419,154 +402,78 @@ public void run() {
...
             @Override
             public void onFailure(@NotNull Call call, @NotNull IOException e) {
-
+                System.out.println("this is fail");
```
Comment (ouyuran) [inline app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java]: What is fail.

File: app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java [hunk 5]
```diff
@@ -419,154 +402,78 @@ public void run() {
...
-                if( ! statusList.get(j).equals("Done")){
-                    Dictionary Info = new Hashtable();
-                    Info.put("Type", typeList.get(j));
-                    Info.put("Status", statusList.get(j));
-                    Info.put("Summary", summaryList.get(j));
-                    Info.put("Assignee", assigneeList.get(j));
-                    Info.put("Task", issueList.get(j));
-                    infoList.add(Info);
+                } catch (JSONException e) {
+                    System.out.println("this is error");
```
Comment (ouyuran) [inline app/src/main/java/com/citrix/taskshiftonandroid/MainActivity.java]: Same issue.

**Review by ouyuran — APPROVED** [review]

**Review by xeal3k — APPROVED** [review]

**Pull request merged**